erpnext/support/doctype/maintenance_schedule/maintenance_schedule.py

```diff
@@ -47,28 +47,37 @@ def on_submit(self):
 				self.validate_serial_no(serial_nos, d.start_date)
 				self.update_amc_date(serial_nos, d.end_date)
 
+			no_email_sp = []
 			if d.sales_person not in email_map:
 				sp = frappe.get_doc("Sales Person", d.sales_person)
-				email_map[d.sales_person] = sp.get_email_id()
+				try:
+					email_map[d.sales_person] = sp.get_email_id()
+				except frappe.ValidationError:
+					no_email_sp.append(d.sales_person)
+					
+			if no_email_sp:
+				frappe.msgprint(
+					frappe._("Setting Events to <b>{0}</b>, since the Employee attached to the below Sales Persons does not have a User ID<br>{1}").format(
```

Comment:
Cool, lets leave out the HTML `<b>` `<br>` tags in the message. Not good for translations.